tensorboard/plugins/profile/pod_viewer/details_card/details-card.ts

```diff
@@ -0,0 +1,160 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+namespace pod_viewer_details_card {
+
+Polymer({
+  is: 'details-card',
+  properties: {
+    nodes: {
+      type: Array,
+      notify: true,
+      observer: 'updateCard_',
+    },
+    name: {
+      type: String,
+      value: null,
+    },
+    id: {
+      type: Number,
+    },
+    utilization: {
+      type: Number,
+    },
+    isChannel: {
+      type: Boolean,
+      value: false,
+    },
+    isAllReduce: {
+      type: Boolean,
+      value: false,
+    },
+    hasReplicaGroups: {
+      type: Boolean,
+      value: false,
+    },
+    isStepBreakdown: {
+      type: Boolean,
+      value: false,
+    },
+    stepBreakdownEle: {
+      type: Array,
+    },
+  },
+  /**
+   * Update the details card.
+   */
+  updateCard_: function(nodes) {
+    if (!nodes || nodes.length == 0) return;
+    this.isChannel = false;
+    this.isAllReduce = false;
+    this.isStepBreakdown = false;
+    this.hasReplicaGroups = false;
+    if (nodes[0].channelId) {
+      this.name = 'Channel #';
+      this.id = nodes[0].channelId;
+      this.isChannel = true;
```

Comment:
instead of `updateCard`, can have computed value for `isChannel`, `name`, and others?
Computed properties are semantically correct for derived properties.

Reply:
Done.